src/commands/code-analyzer/rules.ts

```diff
@@ -27,13 +32,20 @@ export default class RulesCommand extends SfCommand<void> {
 		view: Flags.string({
 			summary: getMessage(BundleName.RulesCommand, 'flags.view.summary'),
 			char: 'v',
-			options: ['table', 'detail'] // TODO: Should probably be enum?
+			default: View.TABLE,
+			options: Object.values(View)
 		})
 	};
 
 	public async run(): Promise<void> {
 		const parsedFlags = (await this.parse(RulesCommand)).flags;
-		const action: RulesAction = new RulesAction();
+		const uxDisplay: UxDisplay = new UxDisplay(this);
+		const dependencies: RulesDependencies = {
+			configLoader: new ConfigLoaderImpl(),
+			engineLoader: new EngineLoaderImpl(),
+			viewer: (parsedFlags.view as View) === View.TABLE ? new RuleTableViewer(uxDisplay) : new RuleDetailViewer(uxDisplay)
+		}
+		const action: RulesAction = new RulesAction(dependencies);
```

Comment:
Constructing your action here makes this difficult to test.

Instead, consider adding a overridable createAction method to this class. Then in your test, you would create override of the RulesCommand which overrides the createAction so that you can construct with your stubbed out dependencies.